galley/pkg/fs/fssource_test.go

```diff
@@ -346,3 +343,12 @@ func logChannelOutput(ch chan resource.Event, count int) string {
 	}
 	return strings.TrimSpace(fmt.Sprintf("%v\n", result))
 }
+
+func checkEventOccurs(expectedEvent string, ch chan resource.Event) bool {
+	for event := range ch {
```

Comment:
Wouldn't this block indefinitely if the event doesn't occur?

Reply:
Good catch!  I added a timing to prevent the infinite loop.